solr/core/src/java/org/apache/solr/search/facet/FacetRequest.java

```diff
@@ -302,7 +298,7 @@ public Query createDomainQuery(FacetContext fcontext) throws IOException {
    */
   public static FacetRequest parse(SolrQueryRequest req, Map<String, Object> params) {
     @SuppressWarnings({"rawtypes"})
-    FacetParser parser = new FacetTopParser(req);
+    FacetParser parser = new FacetParser.FacetTopParser(req);
```

Comment:
Seeing this... I think FacetParser ought to have a static method (`newParser`) without exposing FacetTopParser to outside of itself.  Feel free to leave if you think out of scope but IMO now's a great time.